sos/plugins/openstack_instack.py

```diff
@@ -28,6 +30,16 @@ def setup(self):
         self.add_copy_spec("/home/stack/instackenv.json")
         self.add_copy_spec("/home/stack/undercloud.conf")
 
+        # collect undercloud-debug: undercloud-debug.log will be created so we
+        # need to generate it in a tmp.dir, collect it now and then cleanup it
+        tmpdir = gettempdir()
+        self.get_cmd_output_now("su -s /bin/sh -c 'source ~/stackrc && " +
+                                "/usr/libexec/openstack-tripleo/" +
+                                "undercloud-debug' stack",
+                                timeout=600, runat=tmpdir,
+                                suggest_filename="undercloud-debug.log")
+        remove("%s/undercloud-debug.log" % tmpdir)
```

Comment:
And doesn't that `stack` need to be inside the single quotes? In order to be an argument to the script? (as it's written it looks as though it will be interpreted as an argument to the `sh -c`).